contracts/bond/TimeLockMultiRewardBond.sol

```diff
@@ -0,0 +1,258 @@
+     *
+     * @return Time lock in ms.
+     */
+    function rewardTimeLock(address tokens) external view returns (uint256) {
+        require(_hasRegisteredRewards(tokens), "Rewards: no reward tokens");
+
+        return _rewardPool[tokens].timeLock;
+    }
+
+    /**
+     * @notice The amount in ERC20 token of the reward still outstanding.
+     *
+     * @return Time total reward from the given ERC20 contract.
+     */
+    function rewardAmount(address tokens) external view returns (uint256) {
+        require(_hasRegisteredRewards(tokens), "Rewards: no reward tokens");
+
+        return _rewardPool[tokens].amount;
+    }
+
+    /**
+     * @dev Must be called before the claimant debt tokens are burnt, as they are used with total supply in
+     *      calculating the rewards.
+     */
+    function _calculateRewardDebt(
+        address claimant,
+        uint256 claimantDebtTokens,
+        uint256 totalSupply
+    ) internal {
+        require(claimantDebtTokens < totalSupply, "Rewards: too much debt");
+
+        for (uint256 i = 0; i < EnumerableSetUpgradeable.length(_tokens); i++) {
+            address tokens = EnumerableSetUpgradeable.at(_tokens, i);
+            uint256 rewardDebt = (_rewardPool[tokens].amount *
+                claimantDebtTokens) / totalSupply;
+
+            _claimantToRewardPoolDebt[claimant][tokens] = rewardDebt;
+            emit RewardDebt(tokens, claimant, rewardDebt);
+        }
+    }
+
+    /**
+     * @notice Registers ERC20 tokens already transferred to the contract, adding them to the existing reward set.
+     *
+     * @dev Must be done before redemption is allowed.
+     *
+     * @param tokens ERC20 register of the tokens to transfer in and later distribute as rewards.
+     * @param amount total number of ERC20 tokens to use for reward.
+     * @param timeLock ms delay after redemption is allowed, only after which rewards can be claimed.
+     *        Value of zero means rewards can be claimed without delay after redemption is allowed.
+     *        Will overwrite the existing timeLock if there is already a reward from the ERC20 contract.
+     */
+    function _registerReward(
+        address tokens,
+        uint256 amount,
+        uint256 timeLock
+    ) internal whenNotPaused {
+        require(tokens != address(0), "Rewards: address is zero");
+        require(amount > 0, "Rewards: no reward amount");
+
+        emit RegisterReward(tokens, amount, timeLock);
+
+        _rewardPool[tokens].amount += amount;
+        _rewardPool[tokens].timeLock = timeLock;
+
+        require(
+            IERC20Upgradeable(tokens).balanceOf(address(this)) >=
+                _rewardPool[tokens].amount,
+            "Rewards: not enough held"
+        );
+    }
+
+    /**
+     * @notice Overwrites the existing time lock for rewards from a single ERC20.
+     *
+     * @param tokens ERC20 rewards already registered.
+     * @param timeLock ms to lock rewards after redemption is allowed.
```

Comment:
nit: seconds vs miliiseconds? had to google this one lol

Reply:
Are you saying the Linux timestamp is in seconds not milliseconds?